test/e2e/create_test.go

```diff
@@ -128,6 +133,10 @@ func (tc *testContext) waitForWindowsMachines(machineCount int, phase string) er
 			log.Printf("waiting for %d/%d Windows Machines", machineCount-len(machines.Items), machineCount)
 			return false, nil
 		}
+		// When testing Machine deletion, skipping the phase matching as it could quickly change during deletion
+		if phase == "" {
```

Comment:
Please update the function doc to say that `phase` being an empty implies deletion

Reply:
I dont agree with the comment in the code and we should be cautious with the wording around passing in a an empty phase.

If there are 3 machines in the running state and we want to delete a single one, and wait for that to occur, the function call would look like: `tc.waitForWindowsMachines(2, "Running")`, The reason you wouldnt want to enter `""` as the phase, is that in that case we would be cautious against too many machines being deleted, so we'd want to make sure the remaining Machines are still in the `Running` phase, and not transitioned to the deletion phase.

The phase should be empty whenever we truly don't care about the phase, such as in the case in this commit, where the amount of Machines is expected to be zero, and therefore cannot have a phase.

My point here is that `When testing Machine deletion, skipping the phase matching...` is only accurate when scaling down to zero Machines. Instead I'd prefer wording like: `A phase of "" skips the phase check and just waits until the correct amount of Windows Machines exist`

Reply:
In general I agree with @sebsoto however...

>If there are 3 machines in the running state and we want to delete a single one, and wait for that to occur, the function call would look like: tc.waitForWindowsMachines(2, "Running"),

The function the way it is written will not do what you are saying. Not sure if we need this sort of functionality at the moment.

I think the easiest way forward would be to say that if `machineCount` is 0, then `phase` is ignored and that implies we are waiting for Machines to be deleted.

Reply:
> the function the way it is written will not do what you are saying

I'm not able to see what you mean, the function waits until len(machines.Items) == machineCount and then for each machine that was found it verifies the phase is correct.

> I think the easiest way forward would be to say that if machineCount is 0, then phase is ignored and that implies we are waiting for Machines to be deleted.

I agree with that, that functionality was present in the function already so documenting that makes sense.

Reply:
>I'm not able to see what you mean, the function waits until len(machines.Items) == machineCount and then for each machine that was found it verifies the phase is correct.

Take your example , `tc.waitForWindowsMachines(2, "Running")` when there are 3 `Machines` and one Machine is not in the `Running` phase but say in the `Provisioned` phase.
```go
machines, err := tc.client.Machine.Machines(clusterinfo.MachineAPINamespace).List(context.TODO(), metav1.ListOptions{
			LabelSelector: controllers.MachineOSLabel + "=Windows"})
```
Will always return 3. Your example assumes that the 3rd Machine is going to be deleted but that needn't be the case. The unwritten assumption in the function is that we are actually dealing with `Machines` from a `MachineSet` and we sort of expect them to change state as a unit. If we start going outside the boundaries of that assumption, the function will not work as expected.